### PR TITLE
Test intrusive augmented top-down red-black tree

### PR DESCRIPTION
**Pull request by MaskRay**

### CODE REVIEW

File: unit/test_rbtree.c [hunk 1]
```diff
@@ -1,10 +1,13 @@
-#include <r_util.h>
+#include <r_util/r_rbtree.h>
```
Comment (lowlyw) [inline unit/test_rbtree.c]: revert this line, was fixed in https://github.com/radare/radare2/commit/54d43f62eebd2879d145ba3b1a02530e471ff911

File: unit/test_rbtree.c [hunk 2]
```diff
@@ -13,74 +16,137 @@ static void random_iota(int *a, int n) {
 	}
 }
 
-static int cmp_int(void *user, const void *a, const void *b) {
-	return (int)(ptrdiff_t)a - (int)(ptrdiff_t)b;
+struct Node {
+	int key, size;
```
Comment (lowlyw) [inline unit/test_rbtree.c]: define `size` on a new line

```
struct node {
    int key;
    int size;
    RBNode rb;
};
```
for example

File: unit/test_rbtree.c [hunk 3]
```diff
@@ -13,74 +16,137 @@ static void random_iota(int *a, int n) {
...
+};
+
+static void freefn(RBNode *a) {
+	free(container_of(a, struct Node, rb));
+}
+
+static void size(RBNode *a_) {
+	struct Node *a = container_of(a_, struct Node, rb);
+	a->size = 1;
+	for (int i = 0; i < 2; i++)
```
Comment (lowlyw) [inline unit/test_rbtree.c]: please use `{` and `}` for all loops, if, etc.

File: unit/test_rbtree.c [hunk 4]
```diff
@@ -13,74 +16,137 @@ static void random_iota(int *a, int n) {
...
+		black_ = black;
+	} else if (black_ != black) {
+		printf ("error: different black height\n");
+		return false;
+	}
+	return true;
+}
+
+bool check(RBNode *tree) {
+	return check1(tree, 0, 0, true);
```
Comment (lowlyw) [inline unit/test_rbtree.c]: space after `check1` similar with other places here. use r2 style

File: unit/test_rbtree.c [hunk 5]
```diff
@@ -90,47 +156,100 @@ bool test_r_rbtree_insert_delete(void) {
 
 	// Random
 	random_iota(a, N);
-	insert_delete(a, N);
+	insert_delete(a, N, NULL);
+
+	// Increasing
+	for (i = 0; i < N; i++)
```
Comment (lowlyw) [inline unit/test_rbtree.c]: use `{` `}` for loops again. other places in this func too

File: unit/test_rbtree.c [hunk 6]
```diff
@@ -90,47 +156,100 @@ bool test_r_rbtree_insert_delete(void) {
...
-	insert_delete(a, N);
+	insert_delete(a, N, size);
 
 	// Decreasing
 	for (i = 0; i < N; i++)
 		a[i] = N - 1 - i;
-	insert_delete(a, N);
+	insert_delete(a, N, size);
+
+	mu_end;
```
Comment (lowlyw) [inline unit/test_rbtree.c]: undef the `N` at the end of the func if you're going to use a different `N` in other places

File: unit/test_rbtree.c (hunk 5 above)
Comment (MaskRay) [reply unit/test_rbtree.c]: Done

File: unit/test_rbtree.c (hunk 6 above)
Comment (MaskRay) [reply unit/test_rbtree.c]: Done

File: unit/test_rbtree.c (hunk 4 above)
Comment (MaskRay) [reply unit/test_rbtree.c]: Done

File: unit/test_rbtree.c (hunk 3 above)
Comment (MaskRay) [reply unit/test_rbtree.c]: Done

File: unit/test_rbtree.c (hunk 2 above)
Comment (MaskRay) [reply unit/test_rbtree.c]: Done

File: unit/test_rbtree.c (hunk 1 above)
Comment (MaskRay) [reply unit/test_rbtree.c]: Done

**Comment by lowlyw** [comment]
@radare i dont have commit access, please add me. lgtm

**Pull request merged**